<launch>
  <machine name="robot" address="$(env ROBOT)" env-loader="/home/jfstepha/ros_workspace/env.sh" user="jfstepha" timeout="240"/>
  <!-- see http://www.ros.org/wiki/roslaunch/XML/machine -->

  <!-- change this next line to include the launch file for your robot's hardware drivers here -->
  <node machine="robot" pkg="rosserial_python" type="serial_node.py" args="/dev/ttyACM0" name="serial_node" respawn="true"/>
  <param name="robot_description" textfile="$(find yertle)/yertle.urdf" /> 

<!--  original value 15164 --> 
  <rosparam param="ticks_meter">15500</rosparam>
  
   <node machine="robot" pkg="differential_drive" type="diff_tf.py" name="diff_tf">
     <rosparam param="rate">8.0</rosparam>
     <rosparam param="base_width">0.39</rosparam> <!-- 0.38 was measured value -->
  </node>

  <node machine="robot" pkg="differential_drive" type="pid_velocity.py" name="lpid_velocity" output="screen">
      <remap from="wheel" to="lwheel"/>
      <remap from="motor_cmd" to="lmotor_cmd"/>
      <remap from="wheel_vtarget" to="lwheel_vtarget"/>
      <remap from="wheel_vel" to="lwheel_vel"/>
      <rosparam param="Kp">300</rosparam>
      <rosparam param="Ki">3000</rosparam>
      <rosparam param="Kd">0</rosparam>
      <rosparam param="out_min">-255</rosparam>
      <rosparam param="out_max">255</rosparam>
      <rosparam param="rate">50</rosparam>
      <rosparam param="timeout_ticks">20</rosparam>
      <rosparam param="rolling_pts">5</rosparam>
  </node> 
  <node machine="robot" pkg="differential_drive" type="pid_velocity.py" name="rpid_velocity" output="screen">
      <remap from="wheel" to="rwheel"/>
      <remap from="motor_cmd" to="rmotor_cmd"/>
      <remap from="wheel_vtarget" to="rwheel_vtarget"/>
      <remap from="wheel_vel" to="rwheel_vel"/>
      <rosparam param="Kp">300</rosparam>
      <rosparam param="Ki">3000</rosparam>
      <rosparam param="Kd">0</rosparam>
      <rosparam param="out_min">-255</rosparam>
      <rosparam param="out_max">255</rosparam>
      <rosparam param="rate">50</rosparam>
      <rosparam param="timeout_ticks">20</rosparam>
      <rosparam param="rolling_pts">5</rosparam>
  </node> 
<!--    <node pkg="differential_drive" type="virtual_joystick.py" name="virtual_joystick" output="screen"> 
      <rosparam param="~x_min">-0.4</rosparam>
      <rosparam param="~x_max">0.4</rosparam>
      <rosparam param="~r_min">-2.0</rosparam>
      <rosparam param="~r_max">2.0</rosparam>
  </node>
  -->
  
  <node machine="robot" pkg="differential_drive" type="twist_to_motors.py" name="twist_to_motors" output="screen">
    <rosparam param="base_width">0.35</rosparam>
  </node>
  <node machine="robot" name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" >
  <param name="publish_frequency" value="5" />
  </node>
  
<!--   <node name="yertle_dash" pkg="yertle" type="yertle_dash.py" /> -->
  
  <node machine="robot" name="laptop_bat" pkg="yertle" type="laptop_bat.py" />
  <node machine="robot" name="laptop_wifi" pkg="yertle" type="laptop_wifi.py" />
  
</launch>

